src/profile-generator.cc

```diff
@@ -357,7 +395,7 @@ CpuProfile::CpuProfile(const char* title, bool record_samples)
 
 
 void CpuProfile::AddPath(base::TimeTicks timestamp,
-                         const Vector<CodeEntry*>& path, int src_line) {
+                         const Vector<StackEntry>& path, int src_line) {
```

Comment:
Neither Vector nor StackEntry own the instance of CodeEntry. CodeEntry instance is created and destroyed somewhere in other places which are irrelevant to the profiler. So, it is safe to store StackEntry in the Vector.
Regarding efficiency - I am pretty sure there are not a lot of copy operations are done in the Vector, just because the stacks are not deep and filing procedure is quite straight forward. But if we decide to store the pointer to the StackEntry instance, we would need a place for tracking the memory and call delete for these instances in proper place. Currently Vector is doing all of this for us.